types/node.lua

```diff
@@ -271,50 +269,88 @@ function nodetypes.nnode:outputYourself (typesetter, line)
    end
 end
 
-function nodetypes.nnode:toText ()
+--- Returns the text content of the nnode.
+-- Contrary to the parent class, this is the actual text content of the node,
+-- not a text representation of the node.
+--
+-- @treturn string The text content of the nnode.
+function nnode:toText ()
    return self.text
 end
 
-nodetypes.unshaped = pl.class(nodetypes.nnode)
-nodetypes.unshaped.type = "unshaped"
+--- An unshaped node is a text node that has not been shaped yet.
+--
+-- Derived from `nnode`.
+--
+-- Properties is_unshaped is true.
+-- Note that is_nnode is NOT true, as an unshaped node is not a representable node yet.
+--
+-- @type unshaped
 
-function nodetypes.unshaped:_init (spec)
+local unshaped = pl.class(nnode)
+unshaped.type = "unshaped"
+
+--- Constructor
+--
+-- @tparam table spec A table with the properties of the unshaped node.
+
+function unshaped:_init (spec)
    self:super(spec)
    self.width = nil
```

Comment:
See above. The only reason we may have to clear `self.width` is _if_ a width was passed to the super constructor. The box class is too clever for its own.